### PR TITLE
Integration Test Suite

### PR DESCRIPTION
**Pull request by bachdavi**
This PR adds an integration test suite using GitHub secretes to access RAICloud resources.

**Functionality** 
- [X] `exec`
- [X] `exec_async`
- [X] `load_csv`
- [ ] `load_json`
- [ ] `models`
- [ ] `engine`
- [ ] `oauth`
- [ ] `users`

### CODE REVIEW

File: test/integration.jl [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+        poll_until() do
+            get_engine(ctx, engine_name)[:state] == "PROVISIONED"
+        end
+    else
+        engine_name = existing_engine
+    end
+    try
+        f(engine_name)
+    catch
+        rethrow()
```
Comment (NHDaly) [inline test/integration.jl:52]:
```suggestion
```
this is implied by try-finally

File: test/integration.jl [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+# Creates a database and executes `f` with the name of the created database.  Deletes the
+# database when finished. An already existing database can be supplied to improve local
+# iteration times.
+function with_database(f, ctx, engine_name; existing_database=nothing)
+    isnothing(existing_database) &&
+        create_database(ctx, engine_name, engine_name; overwrite=true)
+    try
+        f(engine_name)
+    catch
+        rethrow()
```
Comment (NHDaly) [inline test/integration.jl:74]:
```suggestion
```

**Review by NHDaly — APPROVED** [review]
❤️ amazing. Thank you David! This is a huge improvement! 🎉

File: test/integration.jl [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+                # Test the synchronous path. We expect a response that contains `metadata`,
+                # `problems`, `results` and the `transaction` information.
+                query_string = "x, x^2, x^3, x^4 from x in {1; 2; 3; 4; 5}"
+                resp = exec(CTX, database_name, engine_name, query_string)
+
+                # transaction
+                @test resp["transaction"][:state] == "COMPLETED"
+
+                # metadata
+                @test resp["metadata"][1][:relationId] == "/:output/Int64/Int64/Int64/Int64"
```
[the remark this answers is not in the document]
Comment (bachdavi) [reply test/integration.jl:99]: 👍

**Review by NHDaly — APPROVED** [review]
<3 Thanks David, this is great! I'll merge it now

**Pull request merged**